### DISCUSSION TITLE
performance roundup

### DISCUSSION THREAD

**Discussion by nisbet-hubbard** [neovim/neovim#30270] [General]
### Problem

Whilst I’m keenly aware with Neovim there’s always a tradeoff between features and raw performance, I’m wondering in which areas refactoring has thus far made the code run more efficiently.

#4448 touches on this, but otherwise the only information I’ve seen on the subject is the following comment from justinmk a number of years ago.

> Nvim is faster in some cases.
> - :write is faster because 'nofsync' is the default (swapfiles are still fsync'd during idle, along with other fsync-related logic improvements which make 'fsync' unnecessary)
> - :!, :grep, etc., is 10x faster for output-heavy commands (where writing to the tty is the bottleneck)
> - For slow 'foldmethod' Nvim will feel faster during insert-mode.
> - Since Nvim 0.3.2 the TUI is much smarter about avoiding full redraws, in more cases than Vim.

### Expected behavior

What other PRs can I look at?

**Comment by justinmk** [comment]
We need to start building out the `Performance` section in `:help vim_diff`.

> What other PRs can I look at?

We've been fairly good at labeling stuff: https://github.com/neovim/neovim/labels/performance

But building a narrative about user-perceptible performance improvements is harder, and definitely something we should call out in `:help vim_diff`.
